synapse/notifier.py

```diff
@@ -432,8 +432,9 @@ async def get_events_for(
         If explicit_room_id is set, that room will be polled for events only if
         it is world readable or the user has joined the room.
         """
-        from_token = pagination_config.from_token
-        if not from_token:
+        if pagination_config.from_token:
+            from_token = pagination_config.from_token
+        else:
             from_token = self.event_sources.get_current_token()
```

Comment:
This looks equivalent, but I'm surprised that mypy complained?

Reply:
Me too, I think it might be an edge case due to it crossing a function boundary or something. Possibly due to scoping rules? I spent ages hunting it down :/

Reply:
Not a problem, just confirmation that it is meant to be the same is fine! 🥳